edge-modules/api-proxy-module/src/monitors/certs_monitor.rs

```diff
@@ -0,0 +1,327 @@
+            };
+
+            //Same thing as above but for private key and server cert
+            match cert_monitor.need_to_rotate_server_cert(Utc::now()).await {
+                Ok((need_to_rotate_cert, server_cert, private_key)) => {
+                    if need_to_rotate_cert == true {
+                        //If we have a new cert, we need to write it in file system
+                        let result = utils::write_binary_to_file(
+                            server_cert.as_bytes(),
+                            PROXY_SERVER_CERT_PATH,
+                        );
+                        match result {
+                            Err(err) => panic!("{:?}", err),
+                            Ok(_) => (),
+                        }
+
+                        //If we have a new cert, we need to write it in file system
+                        let result = utils::write_binary_to_file(
+                            private_key.as_bytes(),
+                            PROXY_SERVER_PRIVATE_KEY_PATH,
+                        );
+                        match result {
+                            Err(err) => panic!("{:?}", err),
+                            Ok(_) => (),
+                        }
+                        need_notify = true;
+                    }
+                }
+                Err(err) => {
+                    need_notify = false;
+                    log::error!("Error while trying to get server cert {:?}", err);
+                }
+            };
+
+            if need_notify == true {
+                notify_certs_rotated.notify();
+            }
+        }
+    });
+
+    Ok((monitor_loop, shutdown_handle))
+}
+
+struct CertificateMonitor {
+    module_id: String,
+    generation_id: String,
+    hostname: String,
+    bundle_of_trust_hash: String,
+    work_load_api_client: edgelet_client::WorkloadClient,
+    cert_expiration_date: DateTime<Utc>,
+    validity_days: Duration,
+}
+
+impl CertificateMonitor {
+    pub fn new(
+        module_id: String,
+        generation_id: String,
+        hostname: String,
+        workload_url: String,
+        validity_days: Duration,
+    ) -> Result<Self, Error> {
+        //Create expiry date in the past so cert has to be rotated now.
+        let cert_expiration_date = DateTime::parse_from_rfc3339(EXPIRY_TIME_START_DATE)
+            .context("Error reading Start date")?
+            .with_timezone(&Utc);
+
+        let work_load_api_client = match edgelet_client::workload(&workload_url) {
+            Ok(work_load_api_client) => (work_load_api_client),
+            Err(err) => {
+                return Err(anyhow::anyhow!(format!(
+                    "Could not get workload client {:?}",
+                    err
+                )))
+            }
+        };
+
+        Ok(CertificateMonitor {
+            module_id,
+            generation_id,
+            hostname,
+            bundle_of_trust_hash: String::from(""),
```

Comment:
```suggestion
            bundle_of_trust_hash: String::default(),
```
or maybe even `None`

Reply:
bundle_of_trust_hash: String::default() => I prefer this option, it is straightforward. Fixed!